src/gui/raster/qgssinglebandpseudocolorrendererwidget.cpp

```diff
@@ -335,19 +458,26 @@ void QgsSingleBandPseudoColorRendererWidget::on_mClassifyButton_clicked()
   QList<double>::const_iterator value_it = entryValues.begin();
   QVector<QColor>::const_iterator color_it = entryColors.begin();
 
+  // calculate a reasonable number of decimals to display
+  double maxabs = log10( std::max<double>( std::fabs( max ), std::fabs( min ) ) );
```

Comment:
This will need to use `qAbs` instead of `std::fabs` - otherwise it breaks the build on some platforms